src/mesh/voxel_cloud.rs

```diff
@@ -685,6 +705,100 @@ impl ScalarField {
         }
         self.shrink_to_fit(volume_value_range_self)
     }
+    /// Computes interpolated values of two scalar fields. Prior to
+    /// interpolation, the scalar fields will be recomputed to distance fields.
```

Comment:
I suggest we don't do it this way. The current approach leaves little control for the caller to optimize and does unnecessary work.

Instead I suggest:

- Make the user compute the distance fields manually before passing the scalar fields in,
- Document that this function assumes they have been computed and that otherwise it is garbage-in, garbage-out
- Don't clone the other scalar field

I will review the body of the function again once this has been changed.

Reply:
Are you sure about this? There is another function called `interpolate_to` doing exactly this. The one you refer to is what can be extracted to the UI operation (unless there is some private field use).

Reply:
Hmm, then maybe let's extract it to the operation, in which case we don't need this function at all?